include/CppUTest/UtestMacros.h

```diff
@@ -200,6 +200,29 @@
 #define UNSIGNED_LONGS_EQUAL_LOCATION(expected, actual, text, file, line)\
   { UtestShell::getCurrent()->assertUnsignedLongsEqual((unsigned long)expected, (unsigned long)actual, text, file, line); }
 
+#ifdef CPPUTEST_USE_LONG_LONG
```

Comment:
Regarding the scenarios, I agree that the first is unlikely and the second is what we're trying to cover.

My point is that _within_ the second scenario, there's two further cases:

1. If we have the `#ifdef`, then the user gets an error that the macro doesn't exist.  Confused, they look at the header file, immediately spot that it's inside an `#ifdef` block, and realise that they need to enable it.  They do so, everything now works.

2. If we don't have the `#ifdef`, then they get an error that a long long value can't be assigned to a long long parameter (they'll probably miss at first that it's `cpputest_longlong` rather than a "real" long long).  Confused, they look at the macro definition.  It gives them no hints.  They look at the method declaration.  It gives them no hints.  Finally, they notice the typedef and look at the type definition, then finally notice that it's in an `#ifdef` (or see the nearby comment about being a placeholder) and realise what they need to do.

The former seems less confusing than the latter to me.  Especially since following the latter chain assumes the use of an IDE with a good "go to definition" implementation -- it'll require a lot more hunting around otherwise.

Making the header file very slightly "prettier" doesn't seem like an important consideration to me compared to that.